src/rules/mod.rs

```diff
@@ -226,131 +223,130 @@ impl<'i, 'de: 'i, R: serde::Deserialize<'de>> serde::Deserialize<'de> for CssRul
         A: serde::de::MapAccess<'de>,
       {
         let mut rule_type: Option<CowArcStr<'de>> = None;
-        let mut value: Option<serde::__private::de::Content> = None;
+        let mut value: Option<Self::Value> = None;
         while let Some(key) = map.next_key()? {
           match key {
             Field::Type => {
               rule_type = Some(map.next_value()?);
             }
             Field::Value => {
-              value = Some(map.next_value()?);
+              let rule_type = rule_type.as_ref().ok_or_else(|| serde::de::Error::missing_field("type"))?;
```

Comment:
Is it safe to assume that the type field always comes before the value field? I think that's why it was buffering before.